redfish-core/lib/account_service.hpp

```diff
@@ -1712,31 +1761,76 @@ inline void triggerUnauthenticatedACFUpload(
-                [asyncResp, decodedAcf](const boost::system::error_code ec,
-                                        const bool isACFWindowActive) {
-                if (ec)
+    // Allow ACF upload when D-Bus property allow_unauth_upload is true (aka
+    // Redfish property AllowUnauthACFUpload).
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, decodedAcf](const boost::system::error_code ec,
+                                const std::variant<bool>& allowed) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR
+                << "D-Bus response error reading allow_unauth_upload: " << ec;
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        const bool* allowUnauthACFUpload = std::get_if<bool>(&allowed);
+        if (allowUnauthACFUpload == nullptr)
+        {
+            BMCWEB_LOG_ERROR << "nullptr for allow_unauth_upload";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        if (*allowUnauthACFUpload)
+        {
+            uploadACF(asyncResp, decodedAcf);
+            return;
+        }
+
+        // Allow ACF upload when D-Bus property ACFWindowActive is true
+        // (aka OpPanel function 74).
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, decodedAcf](const boost::system::error_code ec1,
+                                    const std::variant<bool>& retVal) {
+            bool isActive;
+            if (ec1)
+            {
+                BMCWEB_LOG_ERROR << "Failed to read ACFWindowActive property";
+                // The Panel app doesn't run on simulated systems.
+                isActive = false;
+            }
+            else
+            {
+                const bool* isACFWindowActive = std::get_if<bool>(&retVal);
+                if (isACFWindowActive == nullptr)
                 {
-                    BMCWEB_LOG_ERROR
-                        << "Failed to read ACFWindowActive property";
+                    BMCWEB_LOG_ERROR << "nullptr for ACFWindowActive";
                     messages::internalError(asyncResp->res);
                     return;
                 }
+                isActive = *isACFWindowActive;
+            }
 
-                if (isACFWindowActive)
-                {
-                    uploadACF(asyncResp, decodedAcf);
-                    return;
-                }
-                BMCWEB_LOG_ERROR << "ACF window not active";
-                messages::insufficientPrivilege(asyncResp->res);
+            if (isActive)
+            {
+                uploadACF(asyncResp, decodedAcf);
                 return;
-                });
-        }
-    }
+            }
+
+            BMCWEB_LOG_ERROR << "ACF upload not allowed";
+            messages::insufficientPrivilege(asyncResp->res);
+            return;
+            },
+            "com.ibm.PanelApp", "/com/ibm/panel_app",
+            "org.freedesktop.DBus.Properties", "Get", "com.ibm.panel",
```

Comment:
Same for these, would prefer getProperty

Reply:
Done.